### ISSUE TITLE
Hooks not populating relational data

### ISSUE THREAD

**Issue by Ketcap** [strapi/strapi#2561]
**Informations**
- **Node.js version**: 10.14.1
- **NPM version**: 6.4.1
- **Strapi version**: v3.0.0-alpha.17
- **Database**: Mongo (mLab)
- **Operating system**:  macOS Mojave 10.14

**What is the current behavior?**
While using hooks afterSave and afterCreate tried to get created document fully with relational data. But it doesnt give any populated data. Base fields are filled but relational data.

**Steps to reproduce the problem**
``` await strapi.services.MODEL.fetch(result._id) ```
in any model's (need some relation) afterSave or afterCreate hook

**What is the expected behavior?**
``` await strapi.services.event.fetch(result._id)```
should return the data with populated relational data.

**Comment by lauriejim** [comment]
Hello @Ketcap!

When you create a new entry via the `/post` (I will use Post content type with relation with Tag) it call the `./api/post/controller/Post.js` => `create` function and it call  `./api/post/service/Post.js` => `add` function.

```js
  /**
   * Promise to add a/an post.
   *
   * @return {Promise}
   */

  add: async (values) => {
    // Extract values related to relational data.
    const relations = _.pick(values, Post.associations.map(ast => ast.alias));
    const data = _.omit(values, Post.associations.map(ast => ast.alias));

    // Create entry with no-relational data.
    const entry = await Post.create(data);

    // Create relational data and return the entry.
    return Post.updateRelations({ _id: entry.id, values: relations });
  },
```

You can see we separate `relations` and `data`.
Like that we can create first create the entry and then manage relations correctly.

That's why you don't have access to the relations on `beforeCreate` and `afterCreate` function, because relations are not send at this moment.

**Comment by pedrosimao** [comment]
This is actually very much needed. It is probably not difficult to implement...

**Comment by ikesler-tenengroup** [comment]
The same issue when saving content from Admin. And services are not called in this case, so there is no workaround

**Comment by diogopalhais** [comment]
Hey, did you tried fetch and convert to JSON first? i did that and it bring all the info. Hope it helps

```javascript
afterSave: async (model, response, options) => {

      await strapi.api.match.services.match.findOne({id: String(model.id)})
      .then((response) => response.toJSON())
      .then(async(responseJson) => {
        console.log(responseJson)
      })
      .catch(error =>{
        console.log(error)
      })

  },
```

**Comment by ikesler-tenengroup** [comment]
Yes, manually it's obviously working. Btw, this is the way I'm using currently.

**Comment by 422vishal** [comment]
It is not working even manually for me.
